device/include/ScalarRegisterAccessor.h

```diff
@@ -104,6 +104,18 @@ namespace ChimeraTK {
      */
     void writeIfDifferent(UserType newValue, VersionNumber versionNumber = VersionNumber{nullptr});
 
+    /**
+     * Convenience function to set and write new value. The given version number
+     * is only used in case the value differs. If versionNumber == {nullptr}, a new version number is generated only if
+     * the write actually takes place.
+     */
+    void setAndWrite(UserType newValue, VersionNumber versionNumber = VersionNumber{nullptr});
```

Comment:
just have `VersionNumber versionNumber = {}` as second argument. There will always a new version number be needed.